core/src/parse.rs

```diff
@@ -215,9 +216,33 @@ fn parse_while(input: &str) -> IResult<&str, Instr, VerboseError<&str>> {
     Ok((input, Instr::While(reg, body)))
 }
 
-fn try_each(input: &str) -> IResult<&str, Instr, VerboseError<&str>> {
+// TODO: for now throwing away spaces and comments
+// probably want too keep them when we do source mapping
+fn comment(input: &str) -> IResult<&str, &str, VerboseError<&str>> {
+    let (input, _) = preceded(
+        space0,
+        context(
+            "Comment",
+            terminated(
+                char(';'),
+                cut(tuple((take_till(|c| c == '\n'), line_ending))),
+            ),
+        ),
+    )(input)?;
+    Ok((input, ""))
+}
+
+fn comment_or_spaces(input: &str) -> IResult<&str, &str, VerboseError<&str>> {
+    let (input, _) =
+        many0(tuple((space0, many1(alt((comment, line_ending))), space0)))(
+            input,
+        )?;
+    Ok((input, ""))
+}
```

Comment:
Could we combine these two funcs?

Reply:
We could but i think this way makes it easier to see whats going on